### PR TITLE
Natalyj split color shader material

### PR DESCRIPTION
**Pull request by Natalyj**

### CODE REVIEW

File: src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    let up = new Vector3(0, 1, 0);
+    if (getUnsignedDot(normal, up) > this.dotThreshold) {
+      up = new Vector3(1, 0, 0);
+    }
+
+    const zAxis = normal.clone();
+    const xAxis = cross(zAxis, up);
+    const yAxis = cross(zAxis, xAxis);
+
```
Comment (Natalyj) [inline src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts:49]: look from here

File: src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const position = normal
+      .clone()
+      .setLength(Math.abs(distance))
```
Comment (vinneyto) [inline src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts:52]: this line is unnecessary

File: src/demo/natalyj-demo/SplitColorShaderMaterial/createCustomGUI.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  gui: GUI,
+  colorConfig: ColorConfig,
+  material: SplitColorShaderMaterial
+) => {
+  const onNormalChange = () => {
+    material.uniforms.u_basis = { value: material.calculateBasis() };
+  };
+
+  let { planeNormal } = colorConfig;
+  gui.add(planeNormal, 'x', -1, 1).onChange(onNormalChange).name('normal.x');
```
Comment (vinneyto) [inline src/demo/natalyj-demo/SplitColorShaderMaterial/createCustomGUI.ts:35]: use Euler angles to build normal

File: src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts (hunk 2 above)
Comment (Natalyj) [reply src/demo/natalyj-demo/SplitColorShaderMaterial/SplitColorShaderMaterial.ts]: It's awesome!
Done

File: webpack.config.ts [hunk 4]
```diff
@@ -42,6 +42,8 @@ export default {
     rules: [
       { test: /\.ts/, loader: 'ts-loader' },
       { test: /\.glsl|css/, loader: 'raw-loader' },
+      { test: /\.gltf/, loader: 'gltf-webpack-loader' },
+      { test: /\.bin|png|svg|jpg|gif/, loader: 'file-loader' },
```
Comment (Natalyj) [inline webpack.config.ts:46]: Found two working ways to load gltf:
1. through file-loader, but you need to import supplemental files  (bin, png ...) somewhere to GLTFLoader to find them.
2. through special loader, which will load supplemental files automatically.
Chose the second way as there can be a lot of supplemental files and importing them all can be dispiriting.

File: src/demo/natalyj-demo/SplitColorShaderMaterial/useSplitColorShaderMaterial.ts [hunk 5]
```diff
@@ -4,47 +4,65 @@ import {
...
     new Color('red'),
-    normal
+    colorConfig
+  );
+
+  const loader = new GLTFLoader();
+  loader.load(
+    Fox,
+    (gltf) => {
+      scene.add(gltf.scene);
```
Comment (Natalyj) [inline src/demo/natalyj-demo/SplitColorShaderMaterial/useSplitColorShaderMaterial.ts:47]: It doesn't woooork :(

Comment (Natalyj) [reply src/demo/natalyj-demo/SplitColorShaderMaterial/useSplitColorShaderMaterial.ts:53]: It does now )

**Pull request merged**